### ISSUE TITLE
How to wrap reduxifyNavigator into a component

### ISSUE THREAD

**Issue by namnv16** [react-navigation/react-navigation-redux-helpers#45]
Hi everyone,
I am using 
"react-navigation": "^2.3.1",
"react-navigation-redux-helpers": "^2.0.2"

With reduxifyNavigator, I don't know where to add Handling the Hardware Back Button in Android
I try wrap reduxifyNavigator into a compenet like below but it not work:

`
import {
  createStackNavigator,
  NavigationActions
} from 'react-navigation';
import {
  createStore,
  applyMiddleware,
  combineReducers,
} from 'redux';
import { BackHandler } from "react-native";
import {
  reduxifyNavigator,
  createReactNavigationReduxMiddleware,
  createNavigationReducer,
} from 'react-navigation-redux-helpers';
import { Provider, connect } from 'react-redux';
import React from 'react';

const AppNavigator = createStackNavigator(AppRouteConfigs);

const navReducer = createNavigationReducer(AppNavigator);
const appReducer = combineReducers({
  nav: navReducer,
  ...
});

// Note: createReactNavigationReduxMiddleware must be run before reduxifyNavigator
const middleware = createReactNavigationReduxMiddleware(
  "root",
  state => state.nav,
);

const App = reduxifyNavigator(AppNavigator, "root");
class ReduxNavigation extends React.Component {
  componentDidMount() {
    BackHandler.addEventListener("hardwareBackPress", this.onBackPress);
  }

  componentWillUnmount() {
    BackHandler.removeEventListener("hardwareBackPress", this.onBackPress);
  }

  onBackPress = () => {
    const { dispatch, nav } = this.props;
    if (nav.index === 0) {
      return false;
    }

    dispatch(NavigationActions.back());
    return true;
  };

  render() {
    /* more setup code here! this is not a runnable snippet */ 
    return <App />;
  }
}
const mapStateToProps = (state) => ({
  state: state.nav,
});
const AppWithNavigationState = connect(mapStateToProps)(ReduxNavigation);

const store = createStore(
  appReducer,
  applyMiddleware(middleware),
);

class Root extends React.Component {
  render() {
    return (
      <Provider store={store}>
        <AppWithNavigationState />
      </Provider>
    );
  }
}
`

**Comment by Ashoat** [comment]
Your code doesn't even seem to use the component `reduxifyNavigator` returns?

If you think you have an actual bug, please create an [Expo Snack](https://snack.expo.io/) demonstrating the issue, and I can investigate.

**Comment by Ashoat** [comment]
https://github.com/react-navigation/react-navigation-redux-helpers/issues/44#issuecomment-399252347 <--- this might be helpful

**Comment by namnv16** [comment]
@Ashoat sorry for my bad example.
I fixed by follow your comment #44 
my working code
```
// Setup navigator
const MainStack = createStackNavigator(routesConfig);
const Drawer = createDrawerNavigator(
  {
    Main: { screen: MainStack }
  },
  {
    contentComponent: NavigationDrawer
  }
);

export const AppNavigator = createStackNavigator(
  {
    Splash: {
      screen: SplashScreen,
      description: "Loading screen"
    },
    Drawer: {
      screen: Drawer,
      description: "main drawer"
    }
  },
  {
    initialRouteName: "Splash",
    headerMode: "none"
  }
);

// Create middleware and connect
export const appNavigatorMiddleware = createReactNavigationReduxMiddleware(
  "root",
  state => state.nav
);

const ReduxAppNavigator = reduxifyNavigator(AppNavigator, "root");

// create nav component
class ReduxNavigation extends PureComponent {
  componentDidMount() {
    BackHandler.addEventListener("hardwareBackPress", this.onBackPress);
  }

  componentWillUnmount() {
    BackHandler.removeEventListener("hardwareBackPress", this.onBackPress);
  }

  onBackPress = () => {
    const { dispatch, nav } = this.props;
    if (nav.index === 0) {
      return false;
    }

    dispatch(NavigationActions.back());
    return true;
  };

  render() {
    const { dispatch, nav } = this.props;
    return <ReduxAppNavigator dispatch={dispatch} state={nav} />;
  }
}

const mapNavStateProps = state => ({
  state: state.nav
});

export default connect(mapNavStateProps)(ReduxNavigation);
```

**Comment by derren7532123** [comment]
Hi Guys anyone can provide solution for this addreduxhelper:
import React from 'react'
import * as ReactNavigation from 'react-navigation'
import { connect } from 'react-redux'
import MixedNavigation from './MixedNavigation'

// here is our redux-aware our smart component
function ReduxNavigation (props) {
  const { dispatch, nav } = props
  const navigation =({
    dispatch,
    state: nav
  })

  return <MixedNavigation navigation={navigation} />
}

const mapStateToProps = state => ({ nav: state.nav })
export default connect(mapStateToProps)(ReduxNavigation)
